fix: master CI red — TestValidate references removed async_validate method

## Problem

`master` CI has been red since 2026-02-25. The failing job is `Test`; `Lint`, `Pre-commit`, `Hassfest` and `HACS Validation` all pass.

```
E   AttributeError: 'ArcticSpaClient' object has no attribute 'async_validate'
tests/test_api.py:411 (and 417, 430, 436)
4 failed, 34 passed
```

## Root cause

Commit `56ab979` ("refactor: clean up dead code and manifest version placeholder") removed the `async_validate` method from `api.py`, but `tests/test_api.py::TestValidate` (4 tests) still calls `client.async_validate()`. The method is genuinely gone — `config_flow.py` validates the API key by calling `async_get_status()` directly, so the `TestValidate` class is orphaned.

## Fix

Remove the orphaned `TestValidate` class. This matches the intent of `56ab979` (async_validate was deliberately deleted as dead code) and restores green CI, which unblocks every open dependabot PR and the HACS publish (#18).

## Impact

- Master CI returns to green.
- All open dependabot PRs stop showing this pre-existing failure.

--- a/tests/test_api.py
+++ b/tests/test_api.py
@@ -397,46 +397,6 @@ class TestPutRequests:
 
 
 # ---------------------------------------------------------------------------
-# async_validate
-# ---------------------------------------------------------------------------
-
-
-class TestValidate:
-    """Tests for async_validate."""
-
-    @pytest.mark.asyncio
-    async def test_validate_success(self):
-        """Successful status fetch returns True."""
-        client = ArcticSpaClient("test_key", session=_make_session(_make_response()))
-        assert await client.async_validate() is True
-
-    @pytest.mark.asyncio
-    async def test_validate_auth_failure_returns_false(self):
-        """ArcticSpaAuthError during validate returns False."""
-        client = ArcticSpaClient("bad_key", session=_make_session(_make_response(status=401)))
-        assert await client.async_validate() is False
-
-    @pytest.mark.asyncio
-    async def test_validate_connection_failure_returns_false(self):
-        """ArcticSpaConnectionError during validate returns False."""
-        session = MagicMock(spec=aiohttp.ClientSession)
-        session.closed = False
-        cm = AsyncMock()
-        cm.__aenter__ = AsyncMock(side_effect=aiohttp.ClientError("unreachable"))
-        cm.__aexit__ = AsyncMock(return_value=False)
-        session.get = MagicMock(return_value=cm)
-
-        client = ArcticSpaClient("test_key", session=session)
-        assert await client.async_validate() is False
-
-    @pytest.mark.asyncio
-    async def test_validate_server_error_returns_false(self):
-        """HTTP 500 during validate returns False."""
-        client = ArcticSpaClient("test_key", session=_make_session(_make_response(status=500)))
-        assert await client.async_validate() is False
-
-
-# ---------------------------------------------------------------------------
 # Session lifecycle
 # ---------------------------------------------------------------------------
 

